src/main/java/fi/otavanopisto/kuntaapi/server/index/StopTimeSearcher.java

```diff
@@ -0,0 +1,133 @@
+  private static final String ORGANIZATION_ID_FIELD = "organizationId";
+  private static final String TRIP_ID_FIELD = "tripId";
+  private static final String STOP_ID_FIELD = "stopId";
+  private static final String DEPARTURE_TIME_FIELD = "departureTime";
+  
+  @Inject
+  private Logger logger;
+  
+  @Inject
+  private IndexReader indexReader;
+  
+  @SuppressWarnings ("squid:S00107")
+  public SearchResult<PublicTransportStopTimeId> searchStopTimes(String organizationId, String tripId, String stopId, Integer depratureTimeOnOrAfter, PublicTransportStopTimeSortBy sortBy, SortDir sortDir, Long firstResult, Long maxResults) {
+    BoolQueryBuilder query = createQuery(organizationId, tripId, stopId, depratureTimeOnOrAfter);
+    String sortField = resolveSortField(sortBy);
+    SortOrder sortOrder = resolveSortOrder(sortDir);
+    
+    return searchStopTimes(query, firstResult, maxResults, sortField, sortOrder);
+  }
+
+  private BoolQueryBuilder createQuery(String organizationId, String tripId, String stopId, Integer depratureTimeOnOrAfter) {
+    BoolQueryBuilder query = boolQuery();
+    
+    if (organizationId != null) {
+      query.must(termQuery(ORGANIZATION_ID_FIELD, organizationId));      
+    }
+    
+    if (tripId != null) {
+      query.must(termQuery(TRIP_ID_FIELD, tripId));      
+    }
+    
+    if (stopId != null) {
+      query.must(termQuery(STOP_ID_FIELD, stopId));      
+    }
+    
+    if (depratureTimeOnOrAfter != null) {
+      query.must(rangeQuery(DEPARTURE_TIME_FIELD).gte(depratureTimeOnOrAfter));
+    }
+    return query;
+  }
+
+  private SortOrder resolveSortOrder(SortDir sortDir) {
+    return sortDir == SortDir.ASC ? SortOrder.ASC : SortOrder.DESC;
+  }
+
+  private String resolveSortField(PublicTransportStopTimeSortBy sortBy) {
+    String sortField = null;
+    if (sortBy != null && sortBy == PublicTransportStopTimeSortBy.DEPARTURE_TIME) {
+      sortField = DEPARTURE_TIME_FIELD;
+    }
+    return sortField;
+  }
+  
+  private SearchResult<PublicTransportStopTimeId> searchStopTimes(QueryBuilder queryBuilder, Long firstResult, Long maxResults, String sortField, SortOrder sortOrder) {
+    if (!indexReader.isEnabled()) {
+      logger.warning("Could not search stop times. Search functions are disabled");
+      return null;
+    }
+    
+    SearchRequestBuilder requestBuilder = indexReader
+        .requestBuilder(TYPE)
+        .storedFields(ORGANIZATION_ID_FIELD, ID_FIELD)
+        .setQuery(queryBuilder);
+    
+    if (firstResult != null) {
+      requestBuilder.setFrom(firstResult.intValue());
+    }
+    
+    if (maxResults != null) {
+      requestBuilder.setSize(maxResults.intValue());
+    }
+    
+    if (sortField != null) {
+      requestBuilder.addSort(sortField, sortOrder);
+    }
+      
+    return new SearchResult<>(getStopTimeIds(indexReader.search(requestBuilder)));
+  }
+  
+  private List<PublicTransportStopTimeId> getStopTimeIds(SearchHit[] hits) {
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png 'Severity: MAJOR') Remove this unused private "getStopTimeIds" method. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.com/coding_rules#rule_key=squid%3AUnusedPrivateMethod)